WordPress/Learn
# Treat standalone lesson and lesson without published course the same

Fixes #2681 

Alternative to https://github.com/WordPress/Learn/pull/2740

The idea here is to fix issues with parts of the Lesson UI that expect a published parent course, by treating lessons like this the same as we do standalone lessons. Content creators can compose the course with modules and lessons, and publish the lessons before the course is published, without things like the 'Exit course' link and course outline showing in the frontend, and resulting in 404s if the parent course is accessed.

| Standard Lesson belonging to Course and Module | Lesson in unpublished course |
|--------|--------|
| ![localhost_8888_lesson_wordpress-and-web-servers_(Desktop) (2)](https://github.com/user-attachments/assets/008b9c56-8302-4497-ae9d-625e21ec4bdd) | ![localhost_8888_lesson_lesson-in-unpublished-course_(Desktop)](https://github.com/user-attachments/assets/f77c7fcd-0723-4563-9528-07381a28f895) | 

### Testing
1. Regression test a standard lesson with a parent course. Whether or not a module is used shouldn't matter.
2. Regression test a standalone lesson (lesson with no assigned course). The only difference with prod should be the addition of the `Exit lesson` link in the top left, which goes to the lessons archive.
3. Create a course with a lesson. Publish the lesson but not the course.
4. View the lesson on the frontend, it should display as if standalone.
